include/seqan3/io/alignment_file/detail.hpp

```diff
@@ -348,6 +348,112 @@ parse_cigar(cigar_input_type && cigar_input)
     return {operations, ref_length, seq_length, sc_begin_count, sc_end_count};
 }
 
+/*!\brief Parses a cigar string into a vector of operation-count pairs (e.g. (M, 3)).
+ * \ingroup alignment_file
+ * \tparam cigar_input_type The type of a single pass input view over the cigar string; must model
+ *                          std::ranges::InputRange.
+ * \param[in]  cigar_input  The single pass input view over the cigar string to parse.
+ *
+ * \returns A tuple of size five containing (1) std::vector of operation-count pairs, e.g. (M, 3), that describe
+ *          the alignment, (2) the aligned reference length, (3) the aligned query sequence length, (4) The number of
+ *          soft clipped bases at the start of the alignment and (5) the number of bases at the end of the alignment.
+ *
+ * \details
+ *
+ * For example, the view over the cigar string "1S4M1D2M2S" will return `{[(M,4), (D,1), (M,2)], 7, 6, 1, 2}`.
+ */
+template <std::ranges::InputRange cigar_input_type>
+std::tuple<std::vector<std::pair<char, size_t>>, size_t, size_t, size_t, size_t>
+parse_binary_cigar(cigar_input_type && cigar_input, size_t n_cigar_op)
+{
+    std::vector<std::pair<char, size_t>> operations{};
+    size_t sc_begin_count{}; // number of soft clipped bases at the beginning
+    size_t sc_end_count{};   // number of soft clipped bases at the end
+    char cigar_op{'\0'};
+    size_t cigar_count{0};
+    size_t ref_length{0}, seq_length{0}; // length of aligned part for ref and query
+    uint32_t o_and_c;
+    static char const * CIGAR_MAPPING = "MIDNSHP=X*******";
+
+    auto result = std::tie(operations, ref_length, seq_length, sc_begin_count, sc_end_count);
+
+    auto update_lengths_fn = [&ref_length, &seq_length, &cigar_op, &cigar_count] ()
+        {
+            if (is_char<'M'>(cigar_op) || is_char<'='>(cigar_op) || is_char<'X'>(cigar_op))
+            {
+                ref_length += cigar_count;
+                seq_length += cigar_count;
+            }
+            else if (is_char<'D'>(cigar_op) || is_char<'N'>(cigar_op))
+            {
+                ref_length += cigar_count;
+            }
+            else if (is_char<'I'>(cigar_op))
+            {
+                seq_length += cigar_count;
+            }
+            else // illegal character
+            {
+                if (is_char<'P'>(cigar_op))
+                    throw format_error{"We do currently not support cigar operation 'P'."};
+                else
+                    throw format_error{std::string{"Illegal cigar operation: "} + std::string{cigar_op}};
+            }
+        };
+
+    if (n_cigar_op == 0) // [[unlikely]]
+        return result;
```

Comment:
alright. I thought to explicitly tie the parameters in the result type because I hoped to reduce possibilities for inconsistencies when I change the result type (especially when using auto as return type). I thought RVO also kicks in :D